src/post_action_executor.rs

```diff
@@ -0,0 +1,29 @@
+use enostr::{FilledKeypair, RelayPool};
+use nostrdb::Note;
+use tracing::info;
+
+use crate::{draft::Drafts, post::NewPost, ui::note::PostAction};
+
+pub struct PostActionExecutor {}
+
+impl PostActionExecutor {
+    pub fn execute<'a>(
+        poster: &FilledKeypair<'_>,
```

Comment:
FilledKeypair is just two pointers, you can just copy it. it doesn't need to be passed by reference